### PR TITLE
[improve docs] Example pallet crate and Basic Example pallet

### PR DESCRIPTION
**Pull request by sacha-l**
This fixes the broken links in the crate level documentation of the Examples crate. It also updates the documentation for the Basic Example pallet by removing the template for documenting a pallet (we now how the [this](https://github.com/paritytech/polkadot-sdk/blob/master/docs/DOCUMENTATION_GUIDELINE.md) to refer to instead).

Note: I found it unnecessary to provide a link to the doc guidelines as I don't think this would be where someone should discover them. I also want to flag some ideas that came while making these minor improvements in [this issue](https://github.com/paritytech/polkadot-sdk-docs/issues/27) (for a subsequent PR) as part of ongoing docs work.

### CODE REVIEW

File: substrate/frame/examples/basic/src/lib.rs [hunk 1]
```diff
@@ -15,258 +15,33 @@
...
 
-//! <!-- markdown-link-check-disable -->
 //! # Basic Example Pallet
 //!
-//! <!-- Original author of paragraph: @gavofyork -->
-//! The Example: A simple example of a FRAME pallet demonstrating
-//! concepts, APIs and structures common to most FRAME runtimes.
-//!
-//! Run `cargo doc --package pallet-example-basic --open` to view this pallet's documentation.
+//! A simple example of a FRAME pallet demonstrating concepts, APIs and structures common to most
```
Comment (kianenigma) [inline substrate/frame/examples/basic/src/lib.rs:20]: It is missing the bit linking to Polkadot?

File: substrate/frame/examples/src/lib.rs [hunk 2]
```diff
@@ -17,24 +17,27 @@
...
 //!
-//! This crate contains examples of FRAME pallets. It is not intended to be used in production.
+//! This crate contains a collection of simple examples of FRAME pallets, demonstrating useful
+//! features in action. It is not intended to be used in production.
 //!
 //! ## Pallets
 //!
-//! - [**`pallet_example_basic`**](./basic): A simple example of a FRAME pallet demonstrating
-//! concepts, APIs and structures common to most FRAME runtimes.
+//! - [`pallet_example_basic`]: This pallet demonstrates concepts, APIs and structures common to
```
Comment (kianenigma) [inline substrate/frame/examples/src/lib.rs:25]: Yes, these links seem to have been broken for a while https://github.com/paritytech/polkadot-sdk/pull/1554#discussion_r1327925822

**Review by kianenigma — APPROVED** [review]
Nice cleanup, thanks!

**Review by juangirini — APPROVED** [review]
> I found it unnecessary to provide a link to the doc guidelines as I don't think this would be where someone should discover them.

Fair enough, at first I thought we should add this link, but that's a good reasoning

**Review by liamaharon — APPROVED** [review]

**Pull request merged**